src/meta/src/stream/stream_manager.rs

```
@@ -65,12 +68,58 @@ impl CreateStreamingJobContext {
     pub fn internal_tables(&self) -> Vec<Table> {
         self.internal_tables.values().cloned().collect()
     }
+}
+
+pub enum CreatingState {
+    Failed { reason: MetaError },
+    Canceling,
+    Created,
+}
+
+struct StreamingJobExecution {
+    id: TableId,
+    shutdown_tx: Option<Sender<CreatingState>>,
+}
+
+impl StreamingJobExecution {
+    fn new(id: TableId, shutdown_tx: Sender<CreatingState>) -> Self {
+        Self {
+            id,
+            shutdown_tx: Some(shutdown_tx),
+        }
+    }
+}
+
+#[derive(Default)]
+struct CreatingStreamingJobInfo {
+    streaming_jobs: Mutex<HashMap<TableId, StreamingJobExecution>>,
+}
+
+impl CreatingStreamingJobInfo {
+    async fn add_job(&self, job: StreamingJobExecution) {
+        let mut jobs = self.streaming_jobs.lock().await;
+        jobs.insert(job.id, job);
+    }
+
+    async fn delete_job(&self, job_id: TableId) {
+        let mut jobs = self.streaming_jobs.lock().await;
+        jobs.remove(&job_id);
+    }
 
-    pub fn internal_table_ids(&self) -> Vec<u32> {
-        self.internal_tables.keys().copied().collect()
+    async fn cancel_jobs(&self, job_ids: Vec<TableId>) {
+        let mut jobs = self.streaming_jobs.lock().await;
+        for job_id in job_ids {
+            if let Some(job) = jobs.get_mut(&job_id) && let Some(shutdown_tx) = job.shutdown_tx.take() {
+                let _ = shutdown_tx.send(CreatingState::Canceling).await.inspect_err(|_| {
+                    tracing::warn!("failed to send canceling state");
```

Should we report this error back to the user?